databases/week2/creatingDatabase.sql

```diff
@@ -0,0 +1,28 @@
+-- Part 4: Creating a database 
+
+-- Creating the database of the class (based on year) enrollend into HYF program
+CREATE database Class_year;
+
+use Class_year;
+
+CREATE table `class`(
+`id` int unsigned NOT NULL auto_increment PRIMARY KEY,
+`name` varchar(255) NOT NULL,
+`begins` datetime NOT NULL,
+`ends` datetime NOT NULL
+) ENGINE=InnoDB DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_unicode_ci;
+
+CREATE table `students`(
+`id` int unsigned NOT NULL auto_increment PRIMARY KEY,
+`name` varchar(255) NOT NULL,
+`email` varchar(255) NOT NULL,
+`phone` varchar(255) NULL,
+`class_id` int unsigned NOT NULL,
+CONSTRAINT `fk_class` foreign key (`class_id`) references `class` (`id`)
+) ENGINE=InnoDB DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_unicode_ci;
+
+CREATE INDEX idx_firstname
+ON students (name);
+
+ALTER TABLE class
+ADD status enum('not-started', 'ongoing', 'finished');
```

Comment:
Here you are supposed to make any kind of your own design of DB which at least have 3 tables & one many to many relations & any intermediate table is also a plus. Give it a try here!